bluemira/equilibria/constraints.py

```diff
@@ -366,6 +366,16 @@ def __len__(self):
         """
         return sum([len(c) for c in self.constraints])
 
```

Comment:
Renamed for now. I think changing the name to 'targets' is good for a future PR, as it can be confusing to explain to new users that the MagneticConstraintSet in constraints.py is actually not used as constraints for the optimisation in the usual sense, but the set of targets for the objective function....

I'd be happier if 'constraints' were strictly used only in the OptimisationProblem sense, and that the field targets are renamed, so we can be clear in our documentation about what we're referring to as 'constraints' without the name clash muddying the waters. Nothing to stop the field targets being used as constraints of course, as you say, but I think they remain conceptually different. Should be a separate PR.